src/language-html/embed/event.js

```diff
@@ -0,0 +1,38 @@
+import htmlEventAttributes from "@prettier/html-event-attributes";
+import { getUnescapedAttributeValue } from "../utils/index.js";
+import { formatAttributeValue, shouldHugJsExpression } from "./utils.js";
+
+export default function printEventAttribute(path, options) {
+  const { node } = path;
+
+  if (!htmlEventAttributes.includes(node.fullName) || options.parentParser) {
+    return;
+  }
+
+  const text = getUnescapedAttributeValue(path.node).trim();
+
+  if (!text.includes("{{")) {
+    return async (textToDoc) => {
+      try {
+        return await formatAttributeValue(
+          text,
+          textToDoc,
+          { parser: "__js_expression" },
+          shouldHugJsExpression,
+        );
+      } catch (error) {
+        // @ts-expect-error -- expected
+        if (error.cause?.code !== "BABEL_PARSER_SYNTAX_ERROR") {
+          throw error;
+        }
+      }
```

Comment:
Which case will break if we remove this?

Reply:
I don't think so, the next branch need this since it's not an expression, but the new parser should be able to parse  all.